wfe2/wfe.go

```diff
@@ -2253,6 +2275,27 @@ func (wfe *WebFrontEndImpl) validateReplacementOrder(ctx context.Context, acct *
 	return replaces, renewalInfo.SuggestedWindow.IsWithin(wfe.clk.Now()), nil
 }
 
+func (wfe *WebFrontEndImpl) validateCertificateProfileName(profile string) error {
+	if profile == "" {
+		// No profile name is specified.
+		return nil
+	}
+	if utf8.RuneCountInString(profile) > 32 {
+		// Profile name will not fit in a VARCHAR(32) column.
+		return errors.New("profile name is too long")
+	}
+	if len(wfe.certificateProfileNames) == 0 {
+		// No certificate profiles are configured, so any profile name is valid.
+		return nil
+	}
+	if !slices.Contains(wfe.certificateProfileNames, profile) {
+		// The profile name is not in the list of configured profiles.
+		return errors.New("not a valid profile name")
```

Comment:
I think this error message should include their provided profile name as a quoted string, so if someone starts getting this error they can grep their config files for that name.

Reply:
Sounds good to me.